pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1554 @@
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.BasicAuthenticationSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2AccessCodeSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2AccessCodeSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2AccessCodeSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2ApplicationSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2ApplicationSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2ApplicationSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2ImplicitSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2ImplicitSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2ImplicitSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	case *openapi_v2.SecurityDefinitionsItem_Oauth2PasswordSecurity:
+		if nok, err := k.SecuritySchemeProps.FromGnostic(s.Oauth2PasswordSecurity); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		if nok, err := k.VendorExtensible.FromGnostic(s.Oauth2PasswordSecurity.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		return ok, nil
+	}
+	return ok, nil
+}
+
+// Tag
+
+func (k *Tag) FromGnostic(g *openapi_v2.Tag) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	ok = true
+
+	if nok, err := k.TagProps.FromGnostic(g); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	if nok, err := k.VendorExtensible.FromGnostic(g.VendorExtension); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+	return ok, nil
+}
+
+func (k *TagProps) FromGnostic(g *openapi_v2.Tag) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+	ok = true
+	k.Description = g.Description
+	k.Name = g.Name
+
+	if g.ExternalDocs != nil {
+		k.ExternalDocs = &ExternalDocumentation{}
+		if nok, err := k.ExternalDocs.FromGnostic(g.ExternalDocs); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	return ok, nil
+}
+
+// ExternalDocumentation
+
+func (k *ExternalDocumentation) FromGnostic(g *openapi_v2.ExternalDocs) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+	ok = true
+	k.Description = g.Description
+	k.URL = g.Url
+
+	// data loss! g.VendorExtension
+	if len(g.VendorExtension) != 0 {
+		ok = false
```

Comment:
this data loss code path is missing test coverage

Reply:
added a specific test targeting data loss cases